include/circt/Dialect/HW/HWTypeInterfaces.td

```diff
@@ -29,14 +29,12 @@ def FieldIDTypeInterface : TypeInterface<"FieldIDTypeInterface"> {
       Get the sub-type of a type for a field ID, and the subfield's ID. Strip
       off a single layer of this type and return the sub-type and a field ID
       targeting the same field, but rebased on the sub-type.
-    }], "std::pair<circt::hw::FieldIDTypeInterface, uint64_t>",
-    "getSubTypeByFieldID", (ins "uint64_t":$fieldID)>,
 
-    InterfaceMethod<[{
-      Return the final type targeted by this field ID by recursively walking all
-      nested aggregate types. This is the identity function for ground types.
-    }], "circt::hw::FieldIDTypeInterface", "getFinalTypeByFieldID",
```

Comment:
Looks like this method is now only available under `FieldIdImpl::`, is that right?  This could dispatch to that as `methodBody` I think, but basically the idea is this isn't a hook point anymore which makes sense.